lightning/src/ln/peer_handler.rs

```diff
@@ -1951,11 +1951,10 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 			},
 			Some(peer_lock) => {
 				let peer = peer_lock.lock().unwrap();
-				if !peer.handshake_complete() { return; }
-				debug_assert!(peer.their_node_id.is_some());
 				if let Some((node_id, _)) = peer.their_node_id {
 					log_trace!(self.logger, "Handling disconnection of peer {}", log_pubkey!(node_id));
 					self.node_id_to_descriptor.lock().unwrap().remove(&node_id);
```

Comment:
1. Can you assume it was actually removed here (assert the returned value)?
2. In the above `None` block, can you assert it was already absent from this list?